### PR TITLE
Fixing drifting issue while navigating in a map

### LINKED ISSUE #10: [Bug report] Localization shifts when robot rotates
## Description
When the robot rotates while navigating in a map the localization starts drifting. This is visible in RViz, and happens when running the stack with amcl and a map.

## Steps to reproduce
Run the Noah simulation while performing navigation on a known map (amcl and a known map are requirements to reproduce this). Make the robot rotate (publishing to /cmd_vel for example) and notice how the robot and map start drifting apart. Example where it can be seen that the laser scan and robot position do not correspond to the actual location:

image

## Fix
There's an incorrect configuration in amcl. Explanation: There was a bug with the `odom_model_type` used in amcl (see param [here](http://wiki.ros.org/amcl#Parameters), and the "bug" paragraph just below.  To fix the bug they created a new one model "diff-corrected". 

To fix Noah, changing the odom type of the amcl parameterss to "diff-corrected" and the "odom_alpha" values to some lower (as recommended there) fixes the issue

### PR DESCRIPTION
**Pull request by JCarosella**
Closes #10

### CODE REVIEW

File: noah_2dnav/launch/2dnav.launch [hunk 1]
```diff
@@ -8,43 +8,52 @@
...
+      <param name="recovery_alpha_slow" value="0.0"/>
+      <param name="recovery_alpha_fast" value="0.0"/>
+      <param name="initial_pose_x" value="9.89498"/>
+      <param name="initial_pose_y" value="6.147881"/>
+      <param name="initial_pose_a" value="0.0"/>
+      <!-- <param name="initial_cov_xx" value="0.0"/> -->
+      <!-- <param name="initial_cov_yy" value="0.0"/> -->
+      <!-- <param name="initial_cov_aa" value="0.0"/> -->
+      <param name="gui_publish_rate" value="10.0"/>
+      <param name="save_pose_rate" value="0.5"/>
```
Comment (francocipollone) [inline noah_2dnav/launch/2dnav.launch:28]: `save_pose_rate (double, default: 0.5 Hz). Maximum rate (Hz) at which to store the last estimated pose and covariance to the parameter server,`

Was it causing an issue? It is the same as the supposed default

File: noah_2dnav/launch/2dnav.launch [hunk 2]
```diff
@@ -8,43 +8,52 @@
...
+      <param name="use_map_topic" value="false"/>
```
Comment (francocipollone) [inline noah_2dnav/launch/2dnav.launch:29]: aren't we using the map from the topic that is served by map server?

File: noah_2dnav/launch/2dnav.launch [hunk 3]
```diff
@@ -8,43 +8,52 @@
...
+      <param name="resample_interval" value="1"/>
+      <param name="transform_tolerance" value="0.1"/>
+      <param name="recovery_alpha_slow" value="0.0"/>
+      <param name="recovery_alpha_fast" value="0.0"/>
+      <param name="initial_pose_x" value="9.89498"/>
+      <param name="initial_pose_y" value="6.147881"/>
+      <param name="initial_pose_a" value="0.0"/>
+      <!-- <param name="initial_cov_xx" value="0.0"/> -->
+      <!-- <param name="initial_cov_yy" value="0.0"/> -->
+      <!-- <param name="initial_cov_aa" value="0.0"/> -->
```
Comment (francocipollone) [inline noah_2dnav/launch/2dnav.launch:26]: Are these commented lines adding some info somehow? If not I'd delete it.

**Review by francocipollone — APPROVED** [review]
:1st_place_medal: Nice finding the bug

File: noah_2dnav/launch/2dnav.launch (hunk 1 above)
Comment (JCarosella) [reply noah_2dnav/launch/2dnav.launch:28]: This was not part of the bug. While debugging I went through every param listed in the AMCL ros webpage to try and figure out what might have been causing it, and thus I added them here, in order. 

I left them given that I believe it is nice having them all listed as not to miss one

File: noah_2dnav/launch/2dnav.launch (hunk 3 above)
Comment (JCarosella) [reply noah_2dnav/launch/2dnav.launch:26]: Removed

File: noah_2dnav/launch/2dnav.launch (hunk 2 above)
Comment (francocipollone) [reply noah_2dnav/launch/2dnav.launch:26]: Ahhh I didn't know that we amcl is using the service by default. THanks

**Review by francocipollone — APPROVED** [review]

**Pull request merged**